samcli/local/apigw/local_apigw_service.py

```diff
@@ -49,24 +57,28 @@ def __init__(
         :param str event_type: Type of the event. "Api" or "HttpApi"
         :param str payload_format_version: version of payload format
         :param bool is_default_route: determines if the default route or not
+        :param str stack_path: path of the stack the route is located
         """
         self.methods = self.normalize_method(methods)
         self.function_name = function_name
         self.path = path
         self.event_type = event_type
         self.payload_format_version = payload_format_version
         self.is_default_route = is_default_route
+        self.stack_path = stack_path
 
     def __eq__(self, other):
         return (
             isinstance(other, Route)
             and sorted(self.methods) == sorted(other.methods)
             and self.function_name == other.function_name
             and self.path == other.path
+            and self.stack_path == other.stack_path
         )
 
     def __hash__(self):
-        route_hash = hash(self.function_name) * hash(self.path)
+        # stack_path could be empty string, and hash('') = 0. To avoid that, we add a prefix to stack_path
```

Comment:
This raises a good point, multiplying hashes when they can be expected to be 0 is a footgun.